[Bug]: MockKube3ControllersMockTest.testDeploymentController test case is flaky

### Bug Description

Recently I found out that during build the `testDeploymentController` test case failed (but after re-run it was okay). So we should investigate what might be the cause of such a race condition and solve this.
```java
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] io.strimzi.test.mockkube3.MockKube3ControllersMockTest.testDeploymentController
[ERROR]   Run 1: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 2: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 3: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 4: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 5: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 6: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
```
Here is full log from the fail to investigate: [1]

[1] - https://gist.github.com/see-quick/34451c82f12edf865679448355a59e67

### Steps to reproduce

1. Run the build pipeline

### Expected behavior

testDeploymentController test case should not be flaky at all

### Strimzi version

main

### Kubernetes version

1.25.0

### Installation method

_No response_

### Infrastructure

_No response_

### Configuration files and logs

_No response_

### Additional context

_No response_

--- a/mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java
+++ b/mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java
@@ -134,7 +134,7 @@ public class MockKube3ControllersMockTest {
 
         client.apps().deployments().inNamespace(namespace).resource(dep).create();
         
-        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, 
+        TestUtils.waitFor("Wait for deployment to have non-empty status", 100L, 10_000L, 
                 () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null
                         && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null
                         && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus().getObservedGeneration() != null);

--- a/mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java
+++ b/mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java
@@ -133,8 +133,11 @@ public class MockKube3ControllersMockTest {
                 .build();
 
         client.apps().deployments().inNamespace(namespace).resource(dep).create();
-
-        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null);
+        
+        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, 
+                () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null
+                        && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null
+                        && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus().getObservedGeneration() != null);
 
         Deployment createdDeployment = client.apps().deployments().inNamespace(namespace).withName(deploymentName).get();
         assertThat(createdDeployment, is(notNullValue()));

